File: app/src/main/java/com/commit451/gitlab/model/api/FileUploadResponse.java
package com.commit451.gitlab.model.api;

import com.squareup.moshi.Json;

import org.parceler.Parcel;

/**
 * Response when a file is uploaded
 */
@Parcel
public class FileUploadResponse {

    @Json(name = "alt")
    String alt;
    @Json(name = "url")
    String url;
    @Json(name = "is_image")
    boolean isImage;
    @Json(name = "markdown")
    String markdown;

    protected FileUploadResponse() {
        //for json
    }

    public String getAlt() {
        return alt;
    }

    public String getUrl() {
        return url;
    }

    public boolean isImage() {
        return isImage;
    }

    public String getMarkdown() {
        return markdown;
    }
}

File: app/src/main/java/com/commit451/gitlab/api/response/FileUploadResponse.kt
package com.commit451.gitlab.api.response

import android.os.Parcelable
import com.squareup.moshi.Json
import kotlinx.android.parcel.Parcelize

/**
 * Response when a file is uploaded
 */
@Parcelize
data class FileUploadResponse(
    @Json(name = "alt")
    var alt: String? = null,
    @Json(name = "url")
    var url: String? = null,
    @Json(name = "is_image")
    var isImage: Boolean = false,
    @Json(name = "markdown")
    var markdown: String
): Parcelable
